test/generate_ssl_tests.pl

```diff
@@ -43,6 +43,7 @@ sub print_templates {
     # Add the implicit base configuration.
     foreach my $test (@ssltests::tests) {
         $test->{"server"} = { (%ssltests::base_server, %{$test->{"server"}}) };
+	$test->{"server2"} = { (%ssltests::base_server, %{$test->{"server2"}}) };
```

Comment:
I wonder if you could fill in "server2" only if it is explicitly specified (not "undef") in the test, to avoid the useless noise in test configs that don't exercise SNI.

Then, in the template file, check that "server2" is defined before filling in that section, and in the handshake helper... 

Well, in the handshake helper, you could a) try to opportunistically load server2 and ignore failure; or, perhaps more robust, b) add EnableSNI to the test section whenever server2 is present. Then, whenever we're not testing SNI, the second CTX would be NULL.

Or, as another alternative, in the cases where "server2" isn't specified in the test dictionary, make it point to the same config as "server". That still adds one line to each non-SNI test, but at least not an entire section.